### PR TITLE
Refactor TPU requirements file and pin build dependencies

### PR DESCRIPTION
**Pull request by richardsliu**
Replaces https://github.com/vllm-project/vllm/pull/9898

### CODE REVIEW

**Comment by bvrockwell** [comment]
Hi @robertgshaw2-neuralmagic , please see the fix to docs and dockerfile (much cleaner :))

**Comment by bvrockwell** [comment]
Looks great, @richardsliu !!

File: docs/source/getting_started/tpu-installation.rst [hunk 1]
```diff
@@ -113,27 +113,8 @@ Clone the vLLM repository and go to the vLLM directory:
 
     git clone https://github.com/vllm-project/vllm.git && cd vllm
 
-Uninstall the existing `torch` and `torch_xla` packages:
 
-.. code-block:: bash
-
-    pip uninstall torch torch-xla -y
```
Comment (bvrockwell) [inline docs/source/getting_started/tpu-installation.rst:120]: might want to keep this step

**Review by bvrockwell — APPROVED** [review]
Just small nit, might want to keep the uninstall step, otherwise looks good!

**Review by robertgshaw2-redhat — APPROVED** [review]

**Pull request merged**